src/components/select/select-list/select-list.component.tsx

```diff
@@ -236,35 +265,39 @@ const SelectList = React.forwardRef(
       }
     };
 
-    const renderedChildren = items.map((item) => {
-      const { index, start } = item;
-      const child = childrenList[index];
+    // the rangeExtractor above can cause an undefined value to be appended to the return items.
+    // Easiest way to stop that crashing is just to filter it out.
+    const renderedChildren = items
+      .filter((item) => item !== undefined)
+      .map((item) => {
```

Comment:
suggestion (non-blocking): as we're not using `item` anywhere else we could destructure `index` and `start` in the map callback parameter

```suggestion
      .map(({index, start}) => {
```